db/migrate/20220930133418_set_duration_for_milestone_work_packages.rb

```diff
@@ -0,0 +1,29 @@
+class SetDurationForMilestoneWorkPackages < ActiveRecord::Migration[7.0]
+  def up
+    set_duration(:work_packages)
+    set_duration(:work_package_journals)
+  end
+
+  private
+
+  def set_duration(table)
+    execute <<~SQL.squish
+      UPDATE
+        #{table}
+      SET
+        duration = CASE
+                   WHEN start_date IS NULL OR due_date IS NULL
+                   THEN NULL
+                   ELSE
+                     1
```

Comment:
In this case, the `CASE` statement can be omitted. A milestone-typed workpackage should always get a duration of 1, even if its dates are null.

Reply:
Done.